bombastic/indexer/src/indexer.rs

```diff
@@ -67,15 +68,21 @@ pub async fn run(
                                                             }).to_string();
                                                         bus.send(failed_topic, failure.as_bytes()).await?;
                                                     }
+                                                    }
+                                                }
+                                                Err(e) => {
+                                                    tracing::warn!("Error retrieving document event data, ignoring (error: {:?})", e);
                                                 }
                                             }
-                                            Err(e) => {
-                                                tracing::warn!("Error retrieving document event data, ignoring (error: {:?})", e);
-                                            }
+                                        },
+                                        EventType::Delete => {
+                                            let (_, key) = Storage::key_from_event(&data)?;
+                                            writer.as_mut().unwrap().delete_document(index.index().doc_id_to_term(key.as_str()));
```

Comment:
I'd like to see this changed to:

```
writer.as_mut().unwrap().delete_document(key.as_str());
```

Otherwise we will implicitly 'leak' tantivy types outside of the index modules.